### PR TITLE
Improve PR template

### PR DESCRIPTION
**Pull request by lingmann**
## High Level Description

Improve the PR template by making it more clear which items are required and which are not.

## Related Issues

  - [DCOS-12106](https://mesosphere.atlassian.net/browse/DCOS-12106) Update dcos/dcos PR template with contributing rules and train details

## Checklist for all PR's

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: Just updating docs.
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include: 

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by lingmann** [comment]
Please review @amitaekbote @spahl @alberts. I've updated this PR's description so that it also conforms to the new PR template.

**Comment by amitaekbote** [comment]
I like the first section a lot! Looks good to me, as we make changes to the process we can add/modify some sections.

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
+
+## Checklist for all PR's
+
+  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
+  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
+  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)
+
+## Checklist for component/package updates:
+
+If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to above please also include:
```
Comment (adam-mesos) [inline PULL_REQUEST_TEMPLATE.md]: s/to above/to the above/

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
+
+  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
+  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
+  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)
+
+## Checklist for component/package updates:
+
+If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to above please also include: 
+
+  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/mesos_state/compare/3db593a81d3caa18b87468754918ca3771e8cace...7fa7e6edb4d2eecc0ee12a1f109accddf1978953)
```
Comment (adam-mesos) [inline PULL_REQUEST_TEMPLATE.md]: Could you please use an example repo that we actually include in dcos? (mesos_state is deprecated/removed)
Example: 
https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
+  - [ ] Test Results: [link to CI job test results for component]
+  - [ ] Code Coverage (optional): [link to code coverage report]
```
Comment (adam-mesos) [inline PULL_REQUEST_TEMPLATE.md]: s/(optional)/(if available)/ ?

File: PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,21 +1,34 @@
-High level description including:
- - What does this change enable
- - What bugs does this change fix
- - High-Level how things changed
+___
+**What is the review process and when will my changes land?**
 
-# Issues
+All PR's require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).
```
Comment (adam-mesos) [inline PULL_REQUEST_TEMPLATE.md]: Minor nit: I use `PRs` for plural, and `PR's` for possessive

File: PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,21 +1,34 @@
...
 
-If this is a component/package update, include
+What features does this change enable? What bugs does this change fix? High-Level description of how things changed.
 
- - [ ] Change Log from last: <link>
- - [ ] Test Results: <link>
- - [ ] Code Coverage: <link>
+## Related Issues
 
+  - [DCOS-<number>](https://dcosjira.atlassian.net/browse/DCOS-<number>)
```
Comment (adam-mesos) [inline PULL_REQUEST_TEMPLATE.md]: I'd also love to see the JIRA title/summary on this line, so I don't have to hover to see what each one is.
```
[DCOS-<number>](https://dcosjira.atlassian.net/browse/DCOS-<number>) Foo the Bar so it stops Bazzing.
[DCOS-<number>](https://mesosphere.atlassian.net/browse/DCOS-<number) TRACKING: dcosjira-123 Foo the Bar
```

**Comment by lingmann** [comment]
Hi @adam-mesos I've made the changes you suggested and also moved the template instructions to the bottom, so that the important bits are always at the top. Back to you for another round... thanks!

**Comment by lingmann** [comment]
@adam-mesos has approved this in a Slack message to me (he is currently having some 2fa issues).

**Comment by gpaul** [comment]
@lingmann this is much clearer - thanks

**Pull request merged**